code/game/machinery/clonepod.dm

```diff
@@ -406,139 +556,80 @@ GLOBAL_LIST_INIT(cloner_biomass_items, list(\
 
 /obj/machinery/clonepod/screwdriver_act(mob/user, obj/item/I)
 	. = TRUE
-	// These icon states don't really matter since we need to call update_icon() to handle panel open/closed overlays anyway.
 	default_deconstruction_screwdriver(user, null, null, I)
 	update_icon()
 
 /obj/machinery/clonepod/wrench_act(mob/user, obj/item/I)
 	. = TRUE
 	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
 		return
-	if(occupant)
-		to_chat(user, "<span class='warning'>Can not do that while [src] is in use.</span>")
-		return
 	if(anchored)
 		WRENCH_UNANCHOR_MESSAGE
 		anchored = FALSE
-		connected.pods -= src
-		connected = null
 	else
 		WRENCH_ANCHOR_MESSAGE
 		anchored = TRUE
 
+/obj/machinery/clonepod/attack_hand(mob/user)
+	. = ..()
+	ui_interact(user)
+
 /obj/machinery/clonepod/emag_act(user)
-	malfunction()
+	. = ..()
+	eject_clone(TRUE)
 
-/obj/machinery/clonepod/cmag_act(mob/user)
-	if(HAS_TRAIT(src, TRAIT_CMAGGED))
-		return
-	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
-	to_chat(user, "<span class='warning'>A droplet of bananium ooze seeps into the synthmeat storage chamber...</span>")
-	ADD_TRAIT(src, TRAIT_CMAGGED, CLOWN_EMAG)
-
-/obj/machinery/clonepod/proc/update_clone_antag(mob/living/carbon/human/H)
-	// Check to see if the clone's mind is an antagonist of any kind and handle them accordingly to make sure they get their spells, HUD/whatever else back.
-	if(H.mind in SSticker.mode.syndicates)
-		SSticker.mode.update_synd_icons_added()
-	if(H.mind in SSticker.mode.cult)
-		SSticker.mode.update_cult_icons_added(H.mind) // Adds the cult antag hud
-		SSticker.mode.add_cult_actions(H.mind) // And all the actions
-		if(SSticker.mode.cult_risen)
-			SSticker.mode.rise(H)
-			if(SSticker.mode.cult_ascendant)
-				SSticker.mode.ascend(H)
-
-//Put messages in the connected computer's temp var for display.
-/obj/machinery/clonepod/proc/connected_message(message)
-	if((isnull(connected)) || (!istype(connected, /obj/machinery/computer/cloning)))
-		return FALSE
-	if(!message)
-		return FALSE
+/obj/machinery/clonepod/emp_act(severity)
+	if(prob(50))
+		eject_clone(TRUE)
+	..()
 
-	connected.temp = "[name] : [message]"
-	connected.updateUsrDialog()
-	return TRUE
-
-/obj/machinery/clonepod/proc/go_out()
-	countdown.stop()
-	var/turf/T = get_turf(src)
-	if(mess) //Clean that mess and dump those gibs!
-		for(var/i in missing_organs)
-			var/obj/I = i
-			I.forceMove(T)
-		missing_organs.Cut()
-		mess = FALSE
-		new /obj/effect/gibspawner/generic(get_turf(src), occupant)
-		playsound(loc, 'sound/effects/splat.ogg', 50, 1)
-		update_icon()
-		return
+//TGUI
+/obj/machinery/clonepod/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "CloningPod", "Cloning Pod", 500, 500, master_ui, state)
+		ui.open()
```

Comment:
If the TGUI4 PR is implemented before this PR, this will need to be adjusted. This includes moving the sizing to the corresponding JavaScript file and updating `ui_interact` to its new implementation.